Mathlib/CategoryTheory/Sites/Coherent.lean

```diff
@@ -82,4 +83,148 @@ lemma isSheaf_coherent [Precoherent C] (P : Cᵒᵖ ⥤ Type w) :
     rintro B S ⟨α, _, X, π, rfl, hS⟩
     exact h _ _ _ _ hS
 
+
+namespace coherentTopology
+
+variable {C : Type _} [Category C] [Precoherent C]
+
+variable {X : C}
+/--
+For a precoherent category, any sieve that contains an `EffectiveEpiFamily` is a sieve of the
+coherent topology.
+Note: This is one direction of `mem_sieves_iff_hasEffectiveEpiFamily`, but is needed for the proof.
+-/
+theorem mem_sieves_of_hasEffectiveEpiFamily (S : Sieve X) :
+    (∃ (α : Type) (_ : Fintype α) (Y : α → C) (π : (a : α) → (Y a ⟶ X)),
+        EffectiveEpiFamily Y π ∧ (∀ a : α, (S.arrows) (π a)) ) →
+          (S ∈ GrothendieckTopology.sieves (coherentTopology C) X) := by
+  rintro ⟨α, ⟨h, ⟨Y, ⟨π, hπ⟩⟩⟩⟩
+  have h_le : Sieve.generate (Presieve.ofArrows _ π) ≤ S := by
+      rw [Sieve.sets_iff_generate (Presieve.ofArrows _ π) S]
+      apply Presieve.le_of_factorsThru_sieve (Presieve.ofArrows (fun i => Y i) π) S _
+      intro Y g f
+      use Y, 𝟙 Y
```

Comment:
```suggestion
      rintro y g f
      use y, 𝟙 y
```
You have `Y` already and this made the code harder to read (for me at least)

Reply:
Good point, I'll rename it `W` though to keep with using capital end-of-alphabet letters for objects of C.